tests/test_ctadata.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+import tempfile
+import os
+from conftest import upstream_webdav_server, hash_file, generate_random_file
+import ctadata
+
+
+@pytest.mark.timeout(30)
+def test_apiclient_list(testing_download_service):
+    with upstream_webdav_server():
+        r = ctadata.list_dir(
+            "", downloadservice=testing_download_service['url'])
+        print(r)
+
+        ctadata.APIClient.downloadservice = testing_download_service['url']
+
+        with pytest.raises(ctadata.api.StorageException):
+            ctadata.list_dir("fake-temp")
+
+        r = ctadata.list_dir("lst/")
+
+        expected = ['lst/', 'lst/users/']
+
+        assert len(r) == 2
+        for entry in r:
+            assert entry['href'] in expected
```

Comment:
This is shorted and also will not let `['lst/', 'lst/'] through``

```suggestion
        assert set([entry['href'] for entry in r]) == set(['lst/', 'lst/users/'])
```